source/Mlos.Python/mlos/Optimizers/OptimizerBase.py

```diff
@@ -65,24 +78,97 @@ def optimum(self, stay_focused=False) -> Tuple[Point, Point]: # pylint: disable=
         -------
         best_config_point : Point
             Configuration that corresponds to the optimum objective value.
-        best_objective_point : Point
 
-            Best objective value observed so far (corresponding to the best_config_point).
+        best_objective_point : Point
+            Best optimum value as specified by the OptimumDefinition argument (corresponding to the best_config_point).
 
         """
+        assert optimum_definition in OptimumDefinition
+
         features_df, objectives_df = self.get_all_observations()
-        if not len(features_df):
-            raise ValueError("Can't compute optimum before registering any results.")
+        if not len(features_df.index):
+            raise ValueError("Can't compute optimum before registering any observations.")
+
+        if optimum_definition == OptimumDefinition.BEST_OBSERVATION:
+            return self._best_observation_optimum(features_df=features_df, objectives_df=objectives_df)
+        return self._prediction_based_optimum(features_df=features_df, optimum_definition=optimum_definition, alpha=alpha)
+
+
+    @trace()
+    def _best_observation_optimum(self, features_df: pd.DataFrame, objectives_df: pd.DataFrame) -> Tuple[Point, Point]:
+        objective = self.optimization_problem.objectives[0]
+        if objective.minimize:
+            index_of_best = objectives_df[objective.name].idxmin()
+        else:
+            index_of_best = objectives_df[objective.name].idxmax()
+        optimum_value = Point.from_dataframe(objectives_df.loc[[index_of_best]])
+        config_at_optimum = Point.from_dataframe(features_df.loc[[index_of_best]])
+        return config_at_optimum, optimum_value
+
+    @trace()
+    def _prediction_based_optimum(self, features_df: pd.DataFrame, optimum_definition: OptimumDefinition, alpha: float)-> Tuple[Point, Point]:
+        objective = self.optimization_problem.objectives[0]
+
+        # Let's see if we have them cached before recomputing
+        #
+        if self.cached_predictions_for_observations is None:

```

Comment:
Are we validating the cache in some way? Maybe we could at least check the length?

Reply:
Yeah, I think we could also check if the cache index, and observation index are the same. I'll add that.

Reply:
I just removed the cache :) After the last performance fix, it was not that useful anymore and it was a little risky to keep around.